pycqed/tests/test_MeasurementControl.py

```diff
@@ -403,3 +405,31 @@ def test_data_resolution(self):
         dat = self.MC.run('1D_soft')
         x = dat['dset'][:, 0]
         np.testing.assert_array_almost_equal(x, sweep_pts, decimal=5)
+
+    def test_save_exp_metadata(self):
+        metadata_dict = {
+            'intParam': 1,
+            'floatParam': 2.5e-3,
+            'strParam': 'spam',
+            'listParam': [1, 2, 3, 4],
+            'arrayParam': np.array([4e5, 5e5]),
+            'dictParam': {'a': 1, 'b': 2},
+            'tupleParam': (3, 'c')
+        }
+
+        old_a_tools_datadir = a_tools.datadir
+        a_tools.datadir = self.MC.datadir()
+
+        sweep_pts = np.linspace(0, 10, 30)
+        self.MC.set_sweep_function(None_Sweep())
+        self.MC.set_sweep_points(sweep_pts)
+        self.MC.set_detector_function(det.Dummy_Detector_Soft())
+        self.MC.run('test_exp_metadata', exp_metadata=metadata_dict)
+        a = ma.MeasurementAnalysis(label='test_exp_metadata', auto=False)
+
+        a_tools.datadir = old_a_tools_datadir
+
+        loaded_dict = read_dict_from_hdf5(
+            {}, a.data_file['Experimental Data']['Experimental Metadata'])
+
+        np.testing.assert_equal(metadata_dict, loaded_dict)
```

Comment:
I'm not sure if np.testing works well for complex dictionaries. I always have trouble with testing equality of dicts (i.e., it returns True to soon). However, I have mostly used the default pytest assertEqual. Maybe this works exactly as expected. Let me know, then it should be OK.